indico_toolkit/metrics/compare_ground_truth.py

```diff
@@ -10,23 +9,125 @@ class CompareGroundTruth:
     def __init__(self, ground_truth, predictions):
         self.ground_truth: List[dict] = ground_truth
         self.predictions: List[dict] = predictions
-        self.matching_labels: List[str] = None
-        self.not_matching_labels: List[str] = None
+        self.labels: List[str] = None
+        self.all_label_metrics: List[dict] = None
+        self.overall_metrics: List[dict] = None
 
     def _get_labels(self) -> None:
-        ground_truth_labels = []
-        predictions_labels = []
-
-        for pred in self.ground_truth:
-            if not pred["label"] in ground_truth_labels:
-                ground_truth_labels.append(pred["label"])
-
-        for pred in self.predictions:
-            if not pred["label"] in predictions_labels:
-                predictions_labels.append(pred["label"])
-
-        for label in ground_truth_labels:
-            if label in predictions_labels:
-                self.matching_labels.append(label)
-            else:
-                self.not_matching_labels.append(label)
+        labels = []
+
+        for label in self.ground_truth:
+            if not label in labels:
+                labels.append(label)
+
+        for label in self.predictions:
+            if not label in labels:
+                labels.append(label)
+
+        self.labels = labels
+
+    # TODO import this function from toolkit since it already exists
+    def _sequences_overlap(self, x: dict, y: dict) -> bool:
+        """
+        Boolean return value indicates whether or not seqs overlap
+        """
+        return x["start"] < y["end"] and y["start"] < x["end"]
+
+    def _get_precision(self, true_p: int, false_p: int) -> float:
+        if not true_p:
+            precision = 0
+        else:
+            precision = true_p / (true_p + false_p)
+        return precision
+
+    def _get_recall(self, true_p: int, false_n: int) -> float:
+        if not true_p:
+            recall = 0
+        else:
+            recall = true_p / (true_p + false_n)
+        return recall
+
+    def _get_base_metrics(self, label: str) -> dict:
+        true_pos = 0
+        false_neg = 0
+        false_pos = 0
+        recall = 0
+        precision = 0
+
+        if not self.predictions.get(label, False):
+            false_neg = len(self.ground_truth[label])
+        if not self.ground_truth.get(label, False):
+            false_pos = len(self.predictions[label])
+
+        if self.predictions.get(label, False) and self.ground_truth.get(label, False):
```

Comment:
Same comment as above, may even be able to use any() in this case since we're checking the same item exists in both dictionaries